keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -99,48 +101,89 @@ Other identified non-goals are:
 ### API Changes
 
 PodSpec becomes mutable with regards to Container resources requests and
-limits. PodSpec is extended with information of resources allocated on the
-Node for the Pod. PodStatus is extended to show the actual resources applied
+limits. PodStatus is extended to show the resources allocated for and applied
 to the Pod and its Containers.
 
 Thanks to the above:
 * Pod.Spec.Containers[i].Resources becomes purely a declaration, denoting the
   **desired** state of Pod resources,
-* Pod.Spec.Containers[i].ResourcesAllocated (new object, type v1.ResourceList)
-  denotes the Node resources **allocated** to the Pod and its Containers,
-* Pod.Status.ContainerStatuses[i].Resources (new object, type
+* Pod.Status.ContainerStatuses[i].ResourcesAllocated (new field, type
+  v1.ResourceList) denotes the Node resources **allocated** to the Pod and its
+  Containers,
+* Pod.Status.ContainerStatuses[i].Resources (new field, type
   v1.ResourceRequirements) shows the **actual** resources held by the Pod and
   its Containers.
+* Pod.Status.ContainerStatuses[i].Resize (new field, type map[string]string)
+  explains what is happening for a given resource on a given container.
+
+The new `ResourcesAllocated` field represents in-flight resize operations and
+is driven by state kept in the node checkpoint.  Schedulers should user the
+larger of `Spec.Containers[i].Resources` and
+`Status.ContainerStatuses[i].ResourcesAllocated` when considering available
+space on a node.
 
-A new admission controller named 'PodResourceAllocation' is introduced in order
-to limit access to ResourcesAllocated field such that only Kubelet can update
-this field.
+#### Subresource
 
-Additionally, Kubelet is authorized to update PodSpec, and NodeRestriction
-admission plugin is extended to limit Kubelet's update access only to Pod's
-ResourcesAllocated field for CPU and memory resources.
+For alpha, resource changes will be made by updating the pod spec.  For beta
+(or maybe a followup in alpha), a new subresource, /resize, will be defined.
+This subresource could eventually apply to other resources that carry
+PodTemplates, such as Deployments, ReplicaSets, Jobs, and StatefulSets.  This
+will allow users to grant RBAC access to controllers like VPA without allowing
+full write access to pod specs.
+
+The exact API here is TBD.
 
 #### Container Resize Policy
 
 To provide fine-grained user control, PodSpec.Containers is extended with
-ResizePolicy - a list of named subobjects (new object) that supports 'cpu'
-and 'memory' as names. It supports the following policy values:
-* NoRestart - the default value; resize Container without restarting it,
```

Comment:
typo: which policy is the default value?

Reply:
RestartNotRequired. It is called out in the update.